### ISSUE TITLE
Generating a VF with one master

### ISSUE THREAD

**Issue by underware** [googlefonts/fontmake#565]
There are situations where you want to generate a Variable Font with just one master, located at the default of an axis. Fontmake compiles such a VF without any errors. But once you sanitize the VF, you get the errors below (which makes the VF not working in Chrome on Windows). 

Duplicating the master and adding it as an extreme on the axis, solves the problem. But it would be better if this would not be required. From a pragmatic point of view, two identical masers on one axis are identical to one.

```
ERROR: Variations: Axis count mismatch
ERROR: Variations: Failed to parse variation region list
ERROR: HVAR: Failed to parse item variation store
ERROR: Dropping all Variation tables
ERROR: HVAR: Discarding Variations table
ERROR: Table discarded
ERROR: fvar: Discarding Variations table
ERROR: Table discarded
ERROR: gvar: Discarding Variations table
ERROR: Table discarded
```

**Comment by anthrotype** [comment]
hi!
I am actually not sure what the expected output should be when fontmake is asked to produce a variable font but only one master is provided..
There's basically no variation data, so all variation tables would end up empty, right?
I can't think of a use case for that.

**Comment by underware** [comment]
I have a situation where I need exactly this. 
Duplicating the neutral master generates a VF which sanitizes (see below example 2)
It would make sense if the the first example would compile to the same result.

```
    <axes>
        <axis initial="0" default="0" maximum="1000" minimum="0" name="XXXX" tag="XXXX" />
    </axes>
    <sources>
    	<source filename="A.ufo" name="1">
            <location>
                <dimension name="XXXX" xvalue="0" />
            </location>
            <kerning />
            <info />
        </source>
    </sources>
```
	
	
	
```
    <axes>
        <axis initial="0" default="0" maximum="1000" minimum="0" name="XXXX" tag="XXXX" />
    </axes>
    <sources>
    	<source filename="A.ufo" name="1">
            <location>
                <dimension name="XXXX" xvalue="0" />
            </location>
            <kerning />
            <info />
        </source>
        <source filename="A.ufo" name="1">
            <location>
                <dimension name="XXXX" xvalue="1000" />
            </location>
            <kerning />
            <info />
        </source>
    </sources>
```

**Comment by anthrotype** [comment]
the problem seems to be varLib producing an HVAR VarStore whose VariationRegionList has an invalid RegionAxisCount of 0 when provided with a single-master input, whereas the spec says it should be the same number as axisCount in the 'fvar' table.
For some reasons, when you feed varLib with two identical masters, the content of the HVAR VarStore VarData is still empty, but the VarRegionList has a valid axis count == fvar.AxisCount.
Can you please reopen this issue in fonttools issue tracker where i think it belongs?
thanks

**Comment by anthrotype** [comment]
@underware please try rebuild using fonttools from master branch. I changed it so that it doesn't write HVAR and gvar tables when they contain no variation data. The fvar is still present, so the font still advertises itself as "variable" though contains no deltas.

**Comment by underware** [comment]
Yes it is not compiling to a VF which is sanitised. Great! But the compiled font is different to the one, which is based on two identical masters (on one axis). The second one has a HVAR table, the first not. I think both design-spaces (shown above) should compile to identical results, or?

**Comment by anthrotype** [comment]
well, the second one does have an HVAR table but that table doesn't contain any meaningful data.

**Comment by anthrotype** [comment]
Earlier I said this empty HVAR table doesn't contain any meaningful data, however @behdad disagrees

> If font is deemed variable (ie. has fvar, axes..) then if HVAR is missing, one should load glyph to get any possible advance variation. That's extra cost. Empty HVAR should be encoded IMO.

So I decided to revert https://github.com/fonttools/fonttools/pull/1671, until I figure out a simple/clean way to make fonttools output the RegionAxisCount that OTS expects even when VarStore.VarRegionList is empty.

I also proposed a change to OTS to not check RegionAxisCount when VarRegionList is empty (but even if that's accepted it may take a while before it trickles down to browsers).
